File: 1-refresh-token-strategy/0-typescript/src/modules/user/user.controller.ts
import {
    Controller,
    Get,
    Request,
    UseGuards,
} from "@nestjs/common"
import {
    AtGuard,
} from "../../common"

/**
 * Sample protected routes requiring AtGuard / access JWT.
 */
@Controller("users")
export class UserController {
    /**
     * Protected profile route demonstrating AtGuard.
     *
     * @param req — request with user from JWT
     */
    @UseGuards(AtGuard)
    @Get("profile")
    getProfile(@Request() req: { user: { userId: number } }) {
        return {
            message: "Bạn đã truy cập vào khu vực bảo mật!",
            user: {
                userId: req.user.userId,
            },
        }
    }
}

File: 1-refresh-token-strategy/1-java/src/main/java/com/example/refreshdemo/controller/UserController.java
package com.example.refreshdemo.controller;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/users")
public class UserController {
    @GetMapping("/profile")
    public Map<String, Object> getProfile() {
        String userIdStr = (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        Long userId = Long.valueOf(userIdStr);

        Map<String, Object> userMap = new HashMap<>();
        userMap.put("userId", userId);

        Map<String, Object> response = new HashMap<>();
        response.put("message", "Bạn đã truy cập vào khu vực bảo mật!");
        response.put("user", userMap);

        return response;
    }
}
